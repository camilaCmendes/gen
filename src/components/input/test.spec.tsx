import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { Input } from ".";

describe("Input Component", () => {
  it("should render correctly and matches snapshot", () => {
    const { asFragment } = render(<Input label="Test Input" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should update value on user input", () => {
    render(<Input label="Test Input" />);
    const inputElement: HTMLInputElement = screen.getByLabelText("Test Input");

    fireEvent.change(inputElement, { target: { value: "Hello World" } });

    expect(inputElement?.value).toBe("Hello World");
  });

  it("should display error message when provided", () => {
    const errorMessage = "Error: Incorrect input";
    render(<Input label="Test Input" errorMessage={errorMessage} />);

    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });
});
