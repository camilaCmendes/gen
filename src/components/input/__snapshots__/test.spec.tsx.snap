// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-fm6yoz-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
    >
      Test Input
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-hwkq3c-MuiInputBase-root-MuiFilledInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
        id=":r0:"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
